txpool/txpool.go

```diff
@@ -119,9 +243,19 @@ outer:
 				continue outer
 			}
 		}
-		// all inputs still unspent; update proofs
+
 		for i := range txn.Inputs {
-			cau.UpdateOutputProof(&txn.Inputs[i].Parent)
+			// if the input parent is not ephemeral update its proofs.
+			if txn.Inputs[i].Parent.LeafIndex != sunyata.EphemeralLeafIndex {
+				cau.UpdateOutputProof(&txn.Inputs[i].Parent)
+				continue
+			}
+
+			// if the ephemeral parent was confirmed, replace it with a standard
+			// output.
+			if o, exists := confirmedOutputs[txn.Inputs[i].Parent.ID]; exists {
+				txn.Inputs[i].Parent = o
```

Comment:
just occurred to me that we'll have to be careful not to break signatures if we're going to be updating existing transactions. Currently, I think only the `LeafIndex` and `MerkleProof` fields will change here, which is fine because those fields aren't included in the `SigHash`. But I'm considering adding a `ChainIndex` to `OutputID`, to make it easier for e.g. explorers to locate parent transactions. Since an ephemeral output won't know its `ChainIndex`, that field would have to be left blank -- and importantly, it *must* not be included in the signature data, or else we won't be able to replace ephemerals with standard outputs like we do here.

Reply:
Should I add a regression test for `SigHash`?

Reply:
nah, but it might be good to have a more generic test to ensure that the txpool isn't accidentally invalidating transactions when it updates them